### ISSUE TITLE
Flow error - Record update method not found

### ISSUE THREAD

**Issue by Baransu** [facebook/immutable-js#1224]
### What happened

I've run into issue with Records and flow. I have this snippet for creating Records with desired spec
```js
function createRecord<T: Object>(spec: Object): Record<T> {
  const R = Record(spec);
  return new R(spec);
}

type MyType = Record<{ a: number, b: string }>;
const a: MyType = createRecord({ a: 0, b: '' });

console.log(a.update('a', a => a + 2).get('a'));
```
And then when I run flow (v0.45.0) I have following error: 
```shell
28: console.log(a.update('a', a => a + 2).get('a'));
                  ^^^^^^ property `update`. Property not found in
28: console.log(a.update('a', a => a + 2).get('a'));
                ^ Record
```
From what I know Record has update method available, and running this example gives me expected result. How to handle this? Am I missing something?

**Comment by singhshashi** [comment]
This doesn't seem to throw an error for me. See: 

https://flow.org/try/#0PTAEAEDMBsHsHcBQBjWA7AzgF1Ab1AEoCmqATgCYC+oAvKKUQI4CuAlgwBQDkrAtr8ywBDAEbQiXAJQBuRIkjM0yLK3ShkDIViLEy5ADwAVAFygA8iIBWJLAD4OGAA4lTF68smndsCkdt5EUHV0bEJaQhIfcgdnZBlA+iIsZlI0UDQieEIYknjKOSwAT2dQAFlCw2KicO9ffCFTNGZeESJSABpQEVNsUlY0AHNQSltZVEwcBrKKqvCNIi0dSIoOetMABk7u0C4uYfiUENhxADo4AY4hE+ZHckXuIS5OoVp-F4BqUAAmSROBpIeUhkQA

**Comment by rnons** [comment]
@singhshashi it's not the same case. Seems you can require anything on try flow.

**Comment by leebyron** [comment]
Please make sure you're using the latest version of Immutable (v4.0.0-rc.4) - the `update` method was added to flow definitions in v4.

**Comment by donkeybonks** [comment]
@leebyron I have used the latest flow definitions from v9 and I'm getting errors like these

```
> flow check

Error: src/company/selectors/RuntimeProcessSelectors.js:52
 52:   state.updateIn(['processes', 'byKey'], updater);
             ^^^^^^^^ property `updateIn`. Property not found in
 32: export type RootState = RecordOf<RootStateProps>;
                                      ^^^^^^^^^^^^^^ RootStateProps. See: src/company/selectors/RootSelectors.js:32

Error: src/company/selectors/RuntimeSessionSelectors.js:84
 84:   state.updateIn(['session', 'pendingActions'], actions => actions.concat(addActions));
             ^^^^^^^^ property `updateIn`. Property not found in
 32: export type RootState = RecordOf<RootStateProps>;
                                      ^^^^^^^^^^^^^^ RootStateProps. See: src/company/selectors/RootSelectors.js:32
```

see #1484 also, RootState is defined like it was there.

**Comment by donkeybonks** [comment]
Maybe fixed by #1464 .
